### ISSUE TITLE
Keep losing all tasks from the projects

### ISSUE THREAD

**Issue by surendrajat** [alainm23/planify#418]
**Describe the bug**
I marked a task as completed and then I wanted it back so I toggled the `Show Completed` in Project settings. I saw it once and then it was gone. Now I am also unable to see the number of tasks next to that project in the left pane.

**To Reproduce**
1. Go to a 'Project'
2. Mark a task as complete
3. Check Show Completed button
4. If it still shows up.. try deleting the task and Undoing it. Should be gone from Project along with task counter from left.

**Expected behavior**
A completed task should always be stored under same Project and should be visible if Completed Tasks is checked
**Screenshots**
The task is not visible so I can't show.

**Desktop (please complete the following information):**
 - OS or DE: **elementary OS 5.1.5**
 - Version **2.4.2**

**Additional context**
It happened earlier too but I thought it's todoist issue so I disconnected my account and created new offline Projects but the same issue persists.

**Comment by surendrajat** [comment]
Update: All the tasks from that Project are now gone for good.
Not just that, few other tasks also gone from other Projects and I'm now almost where I started. With few empty projects under one folder.

**Comment by BeatLink** [comment]
Could this be related to #396?

**Comment by surendrajat** [comment]
@BeatLink I faced what you did when I disconnected my todoist account. All projects were there but empty so I created tasks again. But after that, I faced this only for few Projects not all, at random.

This was the series of events:

1. I added todoist account for sync.
2. I added labels in planner and the sync got stuck (spinning forever) and no task went to todoist.
3. I disconnected the todoist account and lost all the tasks, just like you but I didn't care as I wanted to remove todoist anyway.
4. I added tasks again and started using it locally.
5. Marked one task as completed by mistake and wanted to restore it and it was gone soon after.
6. After reboot, all other tasks from that project were gone and newly added projects were back in the old folder with no tasks.
7. Now I've deleted that problematic project (leftover of todoist) and created again. It seems fine till now as I can see completed tasks as well.
